Core/Src/CAN/CAN_IDs.c

```diff
@@ -1,99 +1,94 @@
-/*
- * CAN_IDs.c
- *
- *  Created on: Dec 9, 2020
- *      Author: renanmoreira
- */
 #include "CAN/CAN_IDs.h"
 
-CAN_ID_t CAN_ID_map[CAN_ID_QUAN];
-// TODO(renanmoreira): talvez aumentar capacidade se precisar de mais ids de debug
-uint16_t
-    CAN_ID_POS_INTERNAL_ID_MAP[(ECU_CAN_LAST_POPULATED_ID - ECU_CAN_INITIAL_ID + 1) * 4
-                               + 8];
+#include <stdio.h>
+#include <string.h>
 
-uint16_t convert_id_pos_to_index(uint16_t id, uint16_t pos) {
-    return (id - ECU_CAN_INITIAL_ID) * 4 + pos;
+static CAN_var_inf CAN_ID_map[CAN_GENERAL_ID_QUAN];
+
+static datalog_send_t datalog_send_struct[QUANT_RESERVED_ID];
+
+static datalog_send_t aux_datalog_struct;
+
+static uint16_t quant_of_ext_id;
+
+static void populate_datalog_send_struct();
+static void initialize_CAN_var_inf();
+static int compare_function();
```

Comment:
Não deixe o protótipo vazio. Pq deixar vazio significa que a função pode ter nenhum argumento ou todos os argumentos. O correto é sempre colocar, até quando tiver nenhum argumento o certo é colocar (void), mas não esquenta com isso por agr. 

```suggestion
static int compare_function(const void* value1, const void* value2);
```